lib/code_scanning/rubocop/sarif_formatter.rb

```diff
@@ -1,71 +1,47 @@
 # frozen_string_literal: true
-require 'rubocop/formatter/base_formatter'
 require 'json'
-require 'pathname'
+require_relative 'rule'
 
 module CodeScanning
 
   class SarifFormatter < RuboCop::Formatter::BaseFormatter
     def initialize(output, options = {})
       super
-      @sarif = {}
-    end
-
-    def started(_target_files)
-      @sarif['$schema'] = 'https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json'
-      @sarif['version'] = '2.1.0'
+      @sarif = {
+        '$schema' => 'https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json',
+        'version' => '2.1.0'
+      }
       @rules_map = {}
-      @results = []
       @rules = []
+      @results = []
       @sarif['runs'] = [
-        { 'tool' => {
+        {
+          'tool' => {
             'driver' => { 'name' => 'Rubocop', 'rules' => @rules }
```

Comment:
Did you confirm that the `name` matches what you get in a SARIF file generated by the tool?

Reply:
This Sarif formatter code is used to generate the rules and the run sarif too

Reply:
Cool! I somehow thought this was outputing SARIF already. Great that you managed to wrap it!